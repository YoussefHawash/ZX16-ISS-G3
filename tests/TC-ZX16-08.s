.text
.org 0x0000
    j main

.org 0x0020
main:
    # li16 x1, 0x1234 -> x1 = 0x1234 = 4660
    lui  x1, 0x24       # 0x1234 >> 7 = 0x24
    ori  x1, 0x34       # 0x1234 & 0x7F = 0x34

    # clear x2 = 0
    xor  x2, x2

    # not x3 = ~x3 = 0xFFFF
    xori x3, -1

    # negate x4 = -x4 (assume x4 = 5)
    li   x4, 5
    xori x4, -1
    addi x4, 1

    # increment x5 = x5 + 1
    li   x5, 10
    addi x5, 1

    # decrement x6 = x6 - 1
    li   x6, 10
    addi x6, -1

    ecall 0x00A