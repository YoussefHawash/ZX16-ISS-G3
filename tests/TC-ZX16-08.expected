J 32
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
LUI x1, 0x024
ORI x1, 0x34
XOR x2, x2
XORI x3, 0x7f
LI x4, 0x05
XORI x4, 0x7f
ADDI x4, 0x01
LI x5, 0x0a
ADDI x5, 0x01
LI x6, 0x0a
ADDI x6, 0x7f
ECALL 10    # ProgramExit
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0